packages/twenty-server/src/engine/core-modules/billing/services/billing-webhook-customer.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import Stripe from 'stripe';
+import { Repository } from 'typeorm';
+
+import { BillingCustomer } from 'src/engine/core-modules/billing/entities/billing-customer.entity';
+@Injectable()
+export class BillingWebhookCustomerService {
+  protected readonly logger = new Logger(BillingWebhookCustomerService.name);
+  constructor(
+    @InjectRepository(BillingCustomer, 'core')
+    private readonly billingCustomerRepository: Repository<BillingCustomer>,
+  ) {}
+
+  async processStripeCustomerEvent(
+    stripeCustomerId: string,
+    data:
+      | Stripe.CustomerCreatedEvent.Data
+      | Stripe.CustomerDeletedEvent.Data
+      | Stripe.CustomerUpdatedEvent.Data,
+  ) {
+    const stripeMetadataWorkspaceId = data.object.metadata?.workspaceId;
+    const billingCustomer = await this.billingCustomerRepository.findOne({
+      where: { stripeCustomerId },
+    });
+
+    if (!billingCustomer && !stripeMetadataWorkspaceId) {
+      return; //do nothing
+    }
+    if (billingCustomer && !stripeMetadataWorkspaceId) {
+      return billingCustomer.workspaceId; //send stripe to update (stripe customer metadata is not yet up to date)
+    }
+    await this.billingCustomerRepository.upsert(
+      {
+        workspaceId: stripeMetadataWorkspaceId,
+        stripeCustomerId: data.object.id,
+      },
+      {
+        conflictPaths: ['workspaceId', 'stripeCustomerId'],
+        skipUpdateIfNoValuesChanged: true,
+      },
+    );
+  }
+
+  async verifyStripeCustomerMetadata(stripeCustomer: Stripe.Customer) {
```

Comment:
I think this function is not useful and does not do what we expect it to do (it's returning undefined sometimes, sometimes the workspaceId).

We could rename it to: getWorkspaceIdFromBillingCustomer

Reply:
I feel we actually don't need this method, let's just move this logic to the subscription webhook service